hoomd/md/NeighborList.cc

```diff
@@ -1847,64 +1868,64 @@ pybind11::object NeighborList::getPairListPython(uint64_t timestep)
             }
         }
 
-    // pair_list->shrink_to_fit();
+    pair_list->shrink_to_fit();
 
 #ifdef ENABLE_MPI
 
     uint32_t* global_pair_list = nullptr;
-    pybind11::capsule free_when_done;
+    pybind11::capsule delete_when_done;
     unsigned long shape[] = {0, 2}; // first index needs to be set later
     unsigned long stride[] = {2 * sizeof(uint32_t), sizeof(uint32_t)};
 
     if (m_sysdef->isDomainDecomposed())
         {
-        auto n_elem = 2 * (int)pair_list->size();
-        std::vector<int>* global_n_elem;
+        max_n_elem = pair_list->size();
+        // Throw error if n_elem does not fit into an int
+        if (max_n_elem > std::numeric_limits<int>::max() / 2)
+            throw std::runtime_error("The pair list is too large to be communicated with MPI. \
+                                        For the time being, please use the rank local pair list.");
```

Comment:
```suggestion
                                        Please use the rank local pair list.");
```
I know you plan on expanding this to 64 bit MPI builds, but for now lets just mention what they should do.